sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -639,6 +639,26 @@ object RemoveNoopOperators extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Merge case condition expression with same value.
+ */
+object MergeConditionWithValue extends Rule[LogicalPlan] {
```

Comment:
Do you have this rule execution time if it has thousands of branches.

Reply:
this is rule execution time for 2000 branches 
<img width="1316" alt="image" src="https://user-images.githubusercontent.com/29403543/206434948-6146d550-a1e0-4bc4-b410-27f3bbaf794a.png">

Reply:
@wangyum  have any update ?